Leaderboard/routes/users.js

```diff
@@ -1,9 +1,100 @@
-const express = require('express');
+import express from 'express';
+import _ from 'lodash';
+import crypto from 'crypto';
+import config from 'config';
+
+import log from '../logging';
+import {User} from '../database';
+import {asyncWrap} from '../helper';
+
+const {saltLength, iterations, hashLength, encoding} = config.get('Crypto');
 const router = express.Router();
+const patterns = {
+  usernamePattern: '^[A-Za-z]+$',
+  passwordPattern: '^.{12,}$',
+};
+const usernameRegex = new RegExp(patterns.usernamePattern);
+const passwordRegex = new RegExp(patterns.passwordPattern);
+
+/* Utility Functions */
+const setErrors = (user, password, confirm, email) => ({
+  isUserError: user,
+  isPasswordError: password,
+  isConfirmError: confirm,
+  isEmailError: email,
+});
+
+const calculateSaltHash = password => {
```

Comment:
Seems excessively high, could lead to DoS because our server is too busy with calculating hashes to actually serve requests.